src/interface/InputProps.ts

```diff
@@ -0,0 +1,10 @@
+export interface InputProps {
+    type: string
+    className: string
+    maxLength?: number
+    placeHolder?: string
+    inputRule?: (value: string) => boolean
+    inputChange?: (value: string) => void
+    disabled?: boolean
+    defaultState?: string
+}
```

Comment:
c:
- 리액트에서 export해주는 타입들을 사용하셔도 좋습니다.
- `inputChange` 대신 change 이벤트 핸들러 콜백 인자명으로 익숙한 `onChange` 그대로 사용하는 건 어떨까요?

```suggestion
export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
    inputRule?: (value: string) => boolean
}
```

Reply:
기본적으로 제공해주는건 그대로 쓸 수 있었군요..